### PR TITLE
SNOW-409041 test client failover url

### PR DESCRIPTION
**Pull request by sfc-gh-mkeller**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-409041

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am modifying authorization mechanisms
   - [x] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   This PR adds the new client failover connection test

### CODE REVIEW

File: test/conftest.py [hunk 1]
```diff
@@ -14,7 +14,12 @@
 from snowflake.connector import SnowflakeConnection
 from snowflake.connector.telemetry import TelemetryClient, TelemetryData
 
-from . import CLOUD_PROVIDERS, PUBLIC_SKIP_TAGS, running_on_public_ci
+from . import (
+    CLOUD_PROVIDERS,
+    EXTERNAL_SKIP_TAGS,
```
Comment (sfc-gh-dlynch) [inline test/conftest.py:19]: Are these tags going to be necessary if we run the tests against BC sfctest accounts?

File: test/integ/test_connection.py [hunk 2]
```diff
@@ -1060,3 +1060,13 @@ def test_client_prefetch_threads_setting(conn_cnx):
             cur.execute(f"alter session set client_prefetch_threads={new_thread_count}")
             assert cur._result_set.prefetch_thread_num == new_thread_count
         assert conn.client_prefetch_threads == new_thread_count
+
+
+@pytest.mark.aws
+@pytest.mark.external
+def test_client_failover_connection_url(conn_cnx):
+    with conn_cnx("client_failover") as conn:
```
Comment (sfc-gh-dlynch) [inline test/integ/test_connection.py:1068]: This connection is configured using the client_failover parameters? Are those values in this PR or are they defined elsewhere/outside of github?

File: test/conftest.py (hunk 1 above)
Comment (sfc-gh-mkeller) [reply test/conftest.py:19]: We need a way to skip running these tests when we run them from Jenkins, the tag `external` means not to run on Jenkins and conversely `internal` tests will only run on Jenkins. We already had `internal`, I just added logic for `external`

File: test/integ/test_connection.py (hunk 2 above)
Comment (sfc-gh-mkeller) [reply test/integ/test_connection.py:1067]: They are in this PR, they are just encrypted

**Review by sfc-gh-dlynch — APPROVED** [review]
lgtm, thanks!

**Pull request merged**